addons/mail/static/src/core/common/message_model.js

```diff
@@ -17,6 +17,18 @@ export class Message extends Record {
     static id = "id";
     /** @type {Object.<number, import("models").Message>} */
     static records = {};
+    static new(data) {
+        const message = super.new(data);
+        onChange(message, "isEmpty", () => {
+            if (message.isEmpty && message.isStarred) {
+                message.isStarred = false;
+                const starred = this.store.discuss.starred;
+                starred.counter--;
+                starred.messages.delete(message);
+            }
+        });
```

Comment:
I don't like having to do this in JS. The server should itself unstar the message when we edit/"delete" it, and return the starred value accordingly.